internal/manifest/statecheck/state_check.go

```diff
@@ -0,0 +1,88 @@
+package statecheck
+
+import (
+	"context"
+
+	apiappsv1 "k8s.io/api/apps/v1"
+	"k8s.io/cli-runtime/pkg/resource"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+)
+
+type ManagerStateCheck struct {
+	StatefulSetChecker     StatefulSetStateChecker
+	DeploymentStateChecker DeploymentStateChecker
```

Comment:
Can we make those private?

Reply:
I had to export them for the test

Reply:
If we use the NewManagerStateCheck constructor we can make them private. Or is there any reason for creating the manager directly and not via constructor?

Reply:
no no nevermind, I can use the constructor 👍